python/tvm/target/cuda_scope.py

```diff
@@ -0,0 +1,98 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Global configuration/variable scope for cuda target"""
+import tvm
+from tvm._ffi import register_func as _register_func
+from tvm.contrib import nvcc
+
+from .. import nd as _nd
+
+
+class CudaGlobalScope(object):
+    current = None
+
+    def __init__(self):
+        self._old = CudaGlobalScope.current
+        CudaGlobalScope.current = self
+
+        self._old_cuda_target_arch = None
+        self.cuda_target_arch = None
+
+
+CUDA_GLOBAL_SCOPE = CudaGlobalScope()
+CUDA_DEVICE_TYPE = _nd.device("cuda").device_type
```

Comment:
What happen to this line if CUDA is disabled?

Reply:
It doesn't matter whether CUDA is enabled or not. CUDA_DEVICE_TYPE will be set to 2.